core/src/main/java/jenkins/widgets/HistoryPageFilter.java

```diff
@@ -354,14 +366,38 @@ private boolean fitsSearchString(Object data) {
             return true;
         }
 
-        if (data != null) {
-            if (data instanceof Number) {
-                return data.toString().equals(searchString);
+        if (data == null) {
+            return false;
+        }
+
+        if (data instanceof Number) {
+            return data.toString().equals(searchString);
+        } else {
+            if (UserSearchProperty.isCaseInsensitive()) {
```

Comment:
Usage of this class for search in all places still confuses me, but IIRC I've been outvoted in another PR